test/source/tests/runtime_tests.cpp

```diff
@@ -58,6 +69,30 @@ void concurrencpp::tests::test_runtime_constructor() {
     assert_equal(runtime.thread_pool_executor()->max_worker_idle_time(), opts.max_thread_pool_executor_waiting_time);
     assert_equal(runtime.background_executor()->max_concurrency_level(), opts.max_background_threads);
     assert_equal(runtime.background_executor()->max_worker_idle_time(), opts.max_background_executor_waiting_time);
+
+    auto test_runtime_executor = [&thread_started_callback_invocations_num,
+                                  &thread_terminated_callback_invocations_num](std::shared_ptr<executor> executor) {
+        size_t start_thread_started_callback_invocations_num = thread_started_callback_invocations_num;
+        size_t start_thread_terminated_callback_invocations_num = thread_terminated_callback_invocations_num;
+
+        executor
+            ->submit([&thread_started_callback_invocations_num,
+                      &thread_terminated_callback_invocations_num,
+                      start_thread_started_callback_invocations_num,
+                      start_thread_terminated_callback_invocations_num]() {
+                assert_equal(thread_started_callback_invocations_num, start_thread_started_callback_invocations_num + 1);
+                assert_equal(thread_terminated_callback_invocations_num, start_thread_terminated_callback_invocations_num);
+            })
+            .get();
+
+        executor->shutdown();
+        assert_equal(thread_started_callback_invocations_num, start_thread_started_callback_invocations_num + 1);
```

Comment:
I would also test that the thread name is not empty.

Reply:
This is still not added

Reply:
lines 47 and 52